src/main/java/uk/gov/pay/api/model/PaymentWithLinks.java

```diff
@@ -3,10 +3,10 @@
 import java.net.URI;
 import java.util.List;
 
-public class PaymentWithLinks implements PaymentWithLinksJSON {
+public class PaymentWithLinks implements PaymentWithLinksJSON, PaymentJSON {
```

Comment:
Although we are following the _Open/Close_ and _Single Responsibility_ principles here and using a _Decorator pattern_ to extend the `Payment` object, I believe this is a bit over engineered as we now have `PaymentWithLinks` and `PaymentWithSelfLinks` which is with a bit of hindsight are terrible names (although not as bad as celebrity children names), because any other developer reading this code in the future will keep on asking why do we have these names and we are raising our tech debt.

If every payment will have a `_self` link, then I would made the _links with `_self` as part of the `Payment` object with a generic `Links` attribute and that would make use inheritance if we want to use the NextLinks or any other links.